### ISSUE TITLE
Request a resource, not a scope with TokenClient

### ISSUE THREAD

**Issue by zippy1981** [DuendeArchive/IdentityModel#74]
tl;dr: I'd like extension methods for `TokenClient` similar to `RequestClientCredentialsAsync` that request an oauth token for a resource not scope without using extra parameters because that's the common case for apps that use Azure Ad authentication.

# Long versions

I have used AzureAD as an OpenID provider for apps and APIs I have written in ASP.NET Core using the pattern provided by autogenerated visual studio code for Azure Authentication.

When my web apps want to communicate with the API projects they do so via a JWT token from https://login.microsoftonline.com/ via an [AuthenticationContext](https://docs.microsoft.com/en-us/dotnet/api/microsoft.identitymodel.clients.activedirectory.authenticationcontext?view=azure-dotnet). They send the following in the `POST` body.

name|value
-|-
grant_type | client_credentials
client_id | App Id of client in AzureAd App registration (not object id)
**resource** | App Id of api beingconsumed in AzureAd App registration (not object id)
client_secret | key generated for the client app

This sets the `aud` attribute of the resulting token to be the appid of the API. 

However, if I use your `TokenClient` to generate a token like so:

> var tokenResponse = await tokenClient.RequestClientCredentialsAsync(appId);

That gets a token with an aud of `00000002-0000-0000-c000-000000000000`, which according to [this document](https://blogs.msdn.microsoft.com/besidethepoint/2012/10/23/getting-started-with-azure-active-directory/) is the resource id of Microsoft.Azure.ActiveDirectory, so AAD itself I guess?

I can get the proper resource id with 

>var tokenResponse = await tokenClient.RequestClientCredentialsAsync(null, new Dictionary<string, string>{ {"resource", appId}});

Since a common use case for this will probably be consuming apps using Azure AD authentication, and AzureAd authentication generally means requesting access to a resource and not a scope, can we have extenstion methds like: `RequestResurceClientCredentialsAsync` that assume you want a resource, and take the scope as an extra parameter?

**Comment by leastprivilege** [comment]
As you can see here

https://openid.net/certification/

AAD is not a spec compliant implementation and the resource parameter is proprietary. We don’t (want to) support this.
